src/DataProtection/DataProtection/src/KeyManagement/DefaultKeyResolver.cs

```diff
@@ -123,13 +168,14 @@ private bool CanCreateAuthenticatedEncryptor(IKey key)
         // keep trying until we find one that works.
         var unrevokedKeys = allKeys.Where(key => !key.IsRevoked);
         fallbackKey = (from key in (from key in unrevokedKeys
+                                    where !ReferenceEquals(key, preferredDefaultKey) // Don't reconsider it as a fallback
```

Comment:
Why doesn't we want to reused the default key as the fallback key?

Reply:
Because we ruled it out above.  It was unacceptable for some reason, so it doesn't make sense to say "use it anyway, but generate a new key".  In particular, it would be very weird to have it rejected because AKV was unreachable and then chosen because AKV became available during the fallback check.